File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
        int n = nums.size();
        int sum = 0;
        int left =0;
        int right = 0;
        int minLen = INT_MAX;
        while(right<n){
            sum = sum + nums[right];
            
            while(sum>=target){
                minLen = min(minLen, right-left+1);
                sum = sum - nums[left];
                left++;
            }
                
        
            right++;
            
        }
        if(minLen==INT_MAX){
            return 0;
        }
        else{
            return minLen;
        }
        
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        left =0
        right = 0
        sum =  0
        minLen = float('inf')
        while(right<n):
            sum   = sum + nums[right]
            while(sum>=target):
                minLen = min(minLen, right-left+1)
                sum = sum - nums[left]
                left+=1
            else:
                right += 1
        
        if minLen == float('inf'):
            return 0
        else:
            return minLen
        